### PR TITLE
[AIRFLOW-6314] BigQuery - improve BigQueryHook test

### PR DESCRIPTION
**Pull request by TobKed**
Move methods for BigQueryHook under one testings class
Add new tests
Improve mock assertions

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6314
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: tests/gcp/hooks/test_bigquery.py [hunk 1]
```diff
@@ -56,21 +55,108 @@ def test_bigquery_client_creation(
...
-        self.instance.get_pandas_df('select 1')
+    @mock.patch(
+        'airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id',
+        return_value=("CREDENTIALS", "PROJECT_ID",)
+    )
+    @mock.patch("airflow.gcp.hooks.bigquery.BigQueryHook.get_service")
+    def test_bigquery_table_exists_true(self, mock_get_service, mock_get_creds_and_proj_id):
+        project_id = 'bq-project'
+        dataset_id = 'bq_dataset'
+        table_id = 'bq_table_view'
```
Comment (turbaszek) [inline tests/gcp/hooks/test_bigquery.py]: Maybe it's worth to convert those variables to constants? This will reduce boilerplate code ;)

Comment (TobKed) [reply tests/gcp/hooks/test_bigquery.py]: I think it is good idea however I will address it in the another separate PR for the whole file.

Comment (TobKed) [reply tests/gcp/hooks/test_bigquery.py]: @nuclearpinguin WDYT about https://github.com/apache/airflow/pull/6869 ?

Comment (TobKed) [reply tests/gcp/hooks/test_bigquery.py]: @nuclearpinguin PR mentioned in my previous comment was merged, I've rebased and adjusted slightly and now I think it looks like you suggested :)

**Review by turbaszek — APPROVED** [review]

**Pull request merged**